src/couch_mrview_http.erl

```diff
@@ -576,6 +576,9 @@ check_view_etag(Sig, Acc0, Req) ->
 
 
 parse_json(V) when is_list(V) ->
-    ?JSON_DECODE(V);
+    case lists:all(fun is_binary/1, V) of
```

Comment:
specifically, in the lists:foldl, the original code parsed the arguments for each query.

parse_json should not have to defend itself against being called twice, the error is calling twice and we should address that directly.